addons/mail_plugin/tests/test_controller.py

```diff
@@ -29,49 +30,24 @@ def test_enrich_and_create_company(self):
         partner.invalidate_cache()
         self.assertEqual(partner.parent_id, company, "Should change the company of the partner")
 
+    @mock_auth_method_outlook('employee')
     def test_get_partner_blacklisted_domain(self):
-        """Test enrichment on a blacklisted domain.
-
-        Even is the domain is blacklisted, we should not duplicate the company each
-        time a request is made.
-        """
+        """Test enrichment on a blacklisted domain, should return an error."""
         domain = list(iap_tools._MAIL_DOMAIN_BLACKLIST)[0]
 
-        result = self.mock_plugin_partner_get(
-            "Test", "qsd@" + domain,
-            lambda _, __: {
-                "name": "Name",
-                "email": ["contact@" + domain],
-                "iap_information": "test",
-            },
-        )
-
-        first_company_id = result["partner"]["company"]["id"]
-        self.assertTrue(first_company_id and first_company_id > 0)
-        self.assertEqual(result["partner"]["company"]["additionalInfo"]["iap_information"], "test")
-
-        first_company = self.env["res.partner"].browse(first_company_id)
-        self.assertEqual(first_company.name, "Name")
-        self.assertEqual(first_company.email, "contact@" + domain)
-
-        # Test that we do not duplicate the company and that we return the previous one
-        mock_iap_enrich = Mock()
-        result = self.mock_plugin_partner_get("Test", "qsd@" + domain, mock_iap_enrich)
-        self.assertFalse(
-            mock_iap_enrich.called,
-            "We already enriched this company, should not call IAP a second time")
-
-        second_company_id = result["partner"]["company"]["id"]
-        self.assertEqual(first_company_id, second_company_id, "Should not create a new company")
-
-        # But the same blacklisted domain on a different local part
-        # should create a new company (e.g.: asbl_XXXX@gmail.com VS asbl_YYYY@gmail.com)
-        result = self.mock_plugin_partner_get(
-            "Test", "asbl@" + domain,
-            lambda _, domain: {"name": "Name", "email": ["asbl@" + domain]},
-        )
-        second_company_id = result["partner"]["company"]["id"]
-        self.assertNotEqual(first_company_id, second_company_id, "Should create a new company")
+        data = {
+            "id": 0,
+            "jsonrpc": "2.0",
+            "method": "call",
+            "params": {"email": "contact@" + domain, "name": "test"},
+        }
+        result = self.url_open(
```

Comment:
Are you doing real calls ? Because it seems to bypass mocks and stuff like that. We don't want any runbot to depend on calls to IAP, do we ?

Reply:
Ok, this is linked to the fix... but anyway the mock to IAP should always be called somehow, as otherwise you are doing calls (which we don't want), for example if someone does a wrong override / changes your fix, ...

Reply:
for now yes indeed it won't make the call because of my fix, but ok to mock it for safety (I can even check that the method is not called in the test)